File: 953-reverse-only-letters/reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        start = 0
        end = len(s) - 1
        charArray = list(s)
        while start <= end:
            if charArray[start].isalpha():
                if charArray[end].isalpha():
                    temp = charArray[start]
                    charArray[start] = charArray[end]
                    charArray[end] = temp
                    start +=1
                end -=1
            else:
                start +=1
        return ''.join(charArray)

File: 917-reverse-only-letters/reverse-only-letters.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseOnlyLetters = function(s) {
    let start = 0;
    let end = s.length - 1;
    let charArray = s.split('');
    while(start <= end) {
        if(!!charArray[start].match(/[A-Z|a-z]/i)) {
            if(!!charArray[end].match(/[A-Z|a-z]/i)) {
                let temp = charArray[start];
                charArray[start] = charArray[end];
                charArray[end] = temp;
                start++;
            }
            end--;
        } else {
            start++;
        }
    }
    return charArray.join('');
};